### ISSUE TITLE
Implement the IAS15 integrator

### ISSUE THREAD

**Issue by johnwez1** [jobovy/galpy#637]
### Description

Add the integrator described in https://arxiv.org/abs/1409.4779 and widely used in other software libraries. 

I would like to pick this up if nobody is currently working on it. I have an implementation in C of the original Everhart algorithm (RADAU) and should be just a few additions to my current code for IAS15.

**Comment by jobovy** [comment]
Hi,

Thanks for the suggestion and especially for the offer to help! This has been on the [galpy to-do list for a long time](https://github.com/jobovy/galpy/wiki/Possible-galpy-extensions#:~:text=Add%20the%20IAS15%20symplectic%20integrator), so it would be great to add this! Few comments:

* Note that we can't re-use any of rebound's code due to the license being incompatible (GPL vs. BSD here). Sounds like you have an independent implementation, so that wouldn't be a problem.
* In general it's useful to have both a Python and C version, but given that installing the C code is quite straightforward these days, just doing C would be fine (that's what people would be using anyway)
* The `dop853` integrator in C is probably a good reference point for how to hook up the integrator to galpy: https://github.com/jobovy/galpy/blob/a9f03a99d3b82d6d7f851b2e7484df34f685ee52/galpy/util/leung_dop853.c#L231
* Best way to proceed would be for you to fork, make an `ias15` branch, implement as much as you can, and open a Pull Request. I'll then be happy to help with any issues integrating (ha!) the integrator into galpy and with adding test and documentation.

**Comment by johnwez1** [comment]
> * Note that we can't re-use any of rebound's code due to the license being incompatible (GPL vs. BSD here). Sounds like you have an independent implementation, so that wouldn't be a problem.

Yes, I followed the original Everhart paper and I don't believe the sample Fortran code at the end is under any license.
